### DISCUSSION TITLE
Limit scope/visibility of virtual folders

### DISCUSSION THREAD

**Discussion by uorji3** [drakkan/sftpgo#574] [Q&A] [answered]
Hello, I went through the issues and other discussion points but couldn't find a direct answer to my question so I'll ask here.

Imagine a scenario in which I have two users `alice` and `bob`. I want to set up virtual folders for each user using the Google Cloud Storage backend. I was able to set up proper directory permissions so `alice` can only upload/download files within her given "home" directory.

However, I also don't want `alice` to know that there is another user called `bob`. Here is a screenshot using Filezilla.
![image](https://user-images.githubusercontent.com/2802223/136892042-b537ccda-3c0d-4eca-93c6-87f3116ccf0b.png)

In the screenshot, I am connected to the server as `alice` but I can see that there is a `test2.txt` file under `bob` home directory. I can't download the file (due to permissions) but I can see that it exists.

Is there anyway to configure virtual folders to give the users more privacy?

**Comment by drakkan** [comment]
Hi,

you can remove all the permissions on `/demo/home/bob`, or as better alternative you can create separate virtual folders for each user using the same GCS bucket by specifying a key prefix, so each user has its own private part of the bucket. Using a folder template you can easily create many similar folders with a different key prefix

**Reply by uorji3** [reply]
We are likely going to have hundreds of ftp users so I rather not go the route of explicitly denying users given the large number of combinations.

I've created the virtual folder but perhaps I'm configuring it incorrectly

**Reply by drakkan** [reply]
You can create a per-user virtual folder as explained above (all the virtual folders can share the same bucket). If you use REST API to create your users virtual folders can be automatically created using the same request you use to create the user.

If you use the web ui then you can use the template feature

**Reply by uorji3** [reply]
Pardon the repeated questions but I think I'm having trouble understand what the `Home Dir` means in this screenshot:
![image](https://user-images.githubusercontent.com/2802223/136989075-8cf6ad91-8623-42af-8e85-1471a7cd7af4.png)


I can't figure out if this is the local Linux home directory or the remote home directory. Does my configuration look correct to you?

**Comment by drakkan** [comment, marked as answer]
For a cloud based users the home dir (a local directory) is used to store temporary files. If you are using docker or installing SFTPGo from a deb/rpm package you can just leave the home dir empty it will be automatically populated, otherwise use something like `/tmp` or another path where the user running sftpgo can write

**Reply by uorji3** [reply]
Ok. I was able to set things up to my liking. Thank you so much for the project and your responsiveness.

**Reply by drakkan** [reply]
> Ok. I was able to set things up to my liking. Thank you so much for the project and your responsiveness.

Great. If you can please consider a sponsorship, of course it is not required

https://github.com/sponsors/drakkan
